### PR TITLE
Add Payment date, fixed Organization bounty payment

### LINKED ISSUE #896: Investigate keysend payment and failed to update org
Around 9:30am PT, there was a keysend of 6.5 MM sats. 

Paymenthistories status says keysend is true.
But invoicelist says payment status is false.

"owner_pukey" on invoicelists may also be incorrect. It was labeled 0278528dc0726298778aa8b70a696f0297441b3a5ef9b9ecf2d430945ef0c7dd6f:03a6ea2d9ead2120b12bd66292bb4a302c756983dc45dcb2b364b461c66fd53bcb:1099514511361 when really the sender was 02e833b8b07c25c4f8328f4528ffe695f2a86ab48d7ada1c72f129b83158909c97:03a6ea2d9ead2120b12bd66292bb4a302c756983dc45dcb2b364b461c66fd53bcb:1099515559937

Unless "owner_pubkey" is meant to be the receiver?

"Also during this time relay was throwing a generic rpc error connecting to lnd or proxy which may have cause this issue" - Kev

### LINKED ISSUE #898: Org  Roles and Bounty Payment Updates
- An org user should not be able to add a role to another user that they don't have
- Add a Payment date field when a bounty payment is made
- Bounty Balance should be updated when a user tries to pay with the total bounty budget.

### LINKED ISSUE #900: Add date of payment and adjust txn history in org
@ecurrencyhodler I just created a PR for this, please can you create another Issue for Raphael so he returns the date that payments were made and not the date a bounty was created

_Originally posted by @tobi-bams in https://github.com/stakwork/sphinx-tribes/issues/892#issuecomment-1800824437_

### PR DESCRIPTION
**Pull request by elraphty**
## Describe your changes

This PR fixed the Org Bounty Payment issue, and the Assigned Edit bounty issue, and added payment date to the bounties.

## Issue ticket number and link

Closes #898 
Closes #900
Closes #896 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have tested on Chrome and Firefox
- [ ] I have tested on a mobile device

### CODE REVIEW

File: frontend/app/src/helpers/__test__/helpers.spec.ts [hunk 1]
```diff
@@ -5,7 +5,8 @@ import {
   satToUsd,
   calculateTimeLeft,
   toCapitalize,
-  userHasRole
+  userHasRole,
+  isInvoiceExpired
```
Comment (kevkevinpal) [inline frontend/app/src/helpers/__test__/helpers.spec.ts:9]: why did we add this import? is it used in this file?

File: frontend/app/src/people/widgetViews/OrganizationDetails.tsx [hunk 2]
```diff
@@ -90,6 +90,8 @@ const OrganizationDetails = (props: {
     !isOrganizationAdmin && !userHasRole(main.bountyRoles, userRoles, 'ADD ROLES');
   const addWithdrawDisabled =
     !isOrganizationAdmin && !userHasRole(main.bountyRoles, userRoles, 'WITHDRAW BUDGET');
+  const vieStatsDisabled =
```
Comment (kevkevinpal) [inline frontend/app/src/people/widgetViews/OrganizationDetails.tsx:93]: is this a typo?

**Review by kevkevinpal — COMMENTED** [review]
added small comments but might need to take a look at this one more time

**Comment by ecurrencyhodler** [comment]
@tobi-bams now that this has been merged, should we see updated pay dates in txn history now?

**Comment by tobi-bams** [comment]
> @tobi-bams now that this has been merged, should we see updated pay dates in txn history now?

I will have to see what Raphael is returning and make the right change from my end, I will keep you posted.

**Pull request merged**